### PR TITLE
Store ops bboxes in a linear Uint8Array

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
> **Store ops bboxes in a linear Uint8Array**
>
> This PR changes the way we store bounding boxes so that they use less memory and can be more easily shared across threads in the future.
>
> Instead of storing the bounding box and list of dependencies for each operation that renders _something_, we now only store the bounding box of _every_ operation and no dependencies list. The bounding box of each operation covers the bounding box of all the operations affected by it that render something. For example, the bounding box of a `setFont` operation will be the bounding box of all the `showText` operations that use that font.
>
> This affects the debugging experience in pdfBug, since now the bounding box of an operation may be larger than what it renders itself. To help with this, now when hovering on an operation we also highlight (in red) all its dependents. We highlight with white stripes operations that do not affect any part of the page (i.e. with an empty bbox).
>
> To save memory, we now save bounding box x/y coordinates as uint8 rather than float64. This effectively gives us a 256x256 uniform grid that covers the page, which is high enough resolution for the usecase.

Follow up to https://github.com/mozilla/pdf.js/pull/19043.

Due to the more coarse bounding boxes, some tests that previously had a 1-pixel difference on the edge of the partial render now match exactly. We'll need to re-generate the reference images for them.

### CODE REVIEW

File: web/debugger.mjs [hunk 1]
```diff
@@ -20,6 +20,16 @@ for (const key in OPS) {
   opMap[OPS[key]] = key;
 }
 
+// This is copied from src/display/canvas_dependency_tracker.js
+const EMPTY_BBOX = new Uint32Array(new Uint8Array([255, 255, 0, 0]).buffer)[0];
+const BBoxDecoder = {
```
Comment (nicolo-ribaudo) [inline web/debugger.mjs]: This is copy&pasted in a bunch of places. What's the best locations for it, that is accessible in src/display, in web/pdf_page_view, and in the debugger?

**Comment by nicolo-ribaudo** [comment]
issue16127-partial is an actual failure.

File: web/debugger.mjs (hunk 1 above)
Comment (calixteman) [reply web/debugger.mjs]: If you use it only in web/, you can add it in web/ui_utils.js. If you plan to use it in display then you should add it in display/display_utils.js and do something similar to what we've for `getRGB` (just `grep getRGB`) in order to make it usable in web/.

**Comment by nicolo-ribaudo** [comment]
The preview URL doesn't seem to work, but this should be ready.

File: src/display/canvas_dependency_tracker.js [hunk 2]
```diff
@@ -2,10 +2,71 @@ import { Util } from "../shared/util.js";
...
+  maxY(i) {
+    return (this.#coords[i * 4 + 3] + 1) / 256;
+  }
+}
+
+const ensureDebugMetadata = (map, key) => {
+  if (!map) {
+    return undefined;
+  }
+  if (!map.has(key)) {
```
Comment (calixteman) [inline src/display/canvas_dependency_tracker.js]: nit: `let value = map.get(key); if (!value) { value = { ... }; map.set(key, value); } return value;`
This way we win one map operation.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Comment by nicolo-ribaudo** [comment]
Snapshot differences look correct
